format/FormatNexusJungfrauExt.py

```diff
@@ -0,0 +1,123 @@
+import h5py
+import numpy as np
+
+from xfel.util.jungfrau import pad_stacked_format
+
+from dials.array_family import flex
+
+from dxtbx.format.FormatNexus import FormatNexus
+
+"""
+This class can read a Jungfrau image file that includes, in addition to the corrected data,
+the gain maps, and the dark (pedestal) averages and RMS, providing the user access
+to that data.
+This was written to process SWISSFEL image files created using the CCTBX script jf16m_cxigeom2nexus.py
+
+  libtbx.python cctbx_project/xfel/swissfel/jf16m_cxigeom2nexus.py \
+  unassembled_file=X \
+  raw_file=Y \
+  geom_file=Z \
+  pedestal_file=P \
+  gain_file=G \
+  detector_distance=D \
+  include_spectra=True \
+  output_file=O \
+  raw=False \
+  beam_file=B
+
+where X,Y are the corrected,raw hdf5 files obtained at SWISSFEL for a particular run, Z is the crystFEL geom file
+provided by SWISSFEL for the jungfrau, P is the hdf5 file provided by swissfel containing the pedestal data,  G is
+the gainmap hdf5 file provided by SWISSFEL, D is the detector distance in mm, O is an output filename (to be read by
+this format class) and B is the hdf5 file provided by SWISSFEL containing the wavelength spectra measurements (optional)
+"""
+
+
+class FormatNexusJungfrauExt(FormatNexus):
+    @staticmethod
+    def understand(image_file):
+        with h5py.File(image_file, "r") as handle:
+            try:
+                data = handle["/entry/data"]
+                if "pedestal" in data and "gains" in data and "raw" in data:
+                    return True
+            except (KeyError, AttributeError):
+                pass
+        return False
+
+    def _slice_correction_array(self, array):
+        """
+        :param array: numpy array where the slow axis is image number
+        and e.g. array[i] corresponds to the ith nexus data entry
+        :return: array sliced according to the nexus geometry specification
+        """
+        return np.array(
+            [
+                array[:, self._slices[i][0], self._slices[i][1]]
+                for i in range(len(self._slices))
+            ]
+        )
+
+    def _start(self):
+        super(FormatNexusJungfrauExt, self)._start()
```

Comment:
I recently learned that there is neater `super()` syntax in Python3 

https://github.com/cctbx/dxtbx/pull/298#discussion_r571896958

Reply:
Sounds good, super()._start() it is!